service/topology_coordinator.cc

```diff
@@ -2400,7 +2400,7 @@ future<> topology_coordinator::build_coordinator_state(group0_guard guard) {
     co_await _group0.wait_for_all_nodes_to_finish_upgrade(_as);
 
     rtlogger.info("migrating system_auth keyspace data");
-    co_await auth::migrate_to_auth_v2(_sys_ks.query_processor(), _group0.client(),
+    co_await auth::migrate_to_auth_v2(_sys_ks, _group0.client(),
             [this] (abort_source*) { return start_operation();}, _as);
```

Comment:
Nit: when re-doing upgrade during recovery we will not actually migrate any data. I suggest adjusting our log message to indicate that no data was migrated, if it weren't migrated (preferably, adjust it to be consistent with how we report this for service levels).

Reply:
Actually, I see that the PR in its current shape doesn't do that. Shouldn't we skip migrating to auth-v2 here if it succeeded before? Otherwise we are risking old, outdated auth data overwriting the current auth data in the auth v2 tables.